src/models/tracked.players.spec.js

```diff
@@ -0,0 +1,111 @@
+import { TrackedPlayers } from "./tracked.players.js";
+
+describe("tracked.players.js", function () {
+  describe("TrackedPlayers", function () {
+    describe("if nothing is passed as the second parameter when instantiating", function () {
+      beforeEach(function () {
+        this.currentDate = new Date();
+        this.players = new TrackedPlayers("24");
+      });
+
+      it("the property date should be the current date", function () {
+        expect(this.players.date).toEqual(this.currentDate);
+      });
+    });
+
+    describe("if a date is passed as the second parameter when instantiating", function () {
+      beforeEach(function () {
+        this.currentDate = new Date("September 2000");
+        this.twelveHoursInMs = 12 * 60 * 60 * 1000;
+        this.datePlusTwelveHoursInMs =
+          Date.parse(this.currentDate) + this.twelveHoursInMs;
+
+        this.currentDatePlusTwelveHours = new Date(this.datePlusTwelveHoursInMs);
+        this.players = new TrackedPlayers("24", "September 2000");
+      });
+
+      it("the property date should be the current date, plus twelve hours", function () {
+        expect(this.players.date).toEqual(this.currentDatePlusTwelveHours);
+      });
+    });
+
+    describe("if zero is passed in as the player number", function () {
+      beforeEach(function () {
+        this.currentDate = new Date();
+        this.players = new TrackedPlayers("0", this.currentDate);
+      });
+
+      it("the property 'players' should be 0", function () {
+        expect(this.players.players).toBe(0);
+      });
+    });
+
+    describe("if 5473.4 is passed in as the player number", function () {
```

Comment:
I think with this test you want to show, that you cut off the decimals after 1. Then I propose you change the input data accordingly or am I mistaken?

I would really reduce the test to a couple of lines:

```js
  describe("Player numbers are tracked with one decimal precision", function () {
     it("5473.43 is stored as 5473.4", function () {
        expect(new TrackedPlayers("5473.43").players).toBe(5473.4);
      });
    });
```